dynamic_stack_decider/src/dynamic_stack_decider/sequence_element.py

```diff
@@ -0,0 +1,58 @@
+from dynamic_stack_decider.abstract_stack_element import AbstractStackElement
+from dynamic_stack_decider.abstract_action_element import AbstractActionElement
+
+
+class SequenceElement(AbstractStackElement):
+    """
+    A sequence element contains multiple action elements.
+    The sequence element executes all the actions consecutively. That means that the first action will be performed
+    until it pops itself off the stack. Then, the second action is performed. When the last action is popped, the
+    sequence element pops itself too.
+    This is not an abstract class to inherit from.
+    """
+    def __init__(self, blackboard, dsd, actions=()):
+        """
+        :param actions: list of initialized action elements
+        """
+        super(SequenceElement, self).__init__(blackboard, dsd)
+        self.actions = actions
+        self.current_action_index = 0
+
+    def perform(self, reevaluate=False):
+        self.current_action.perform()
+
+    def pop_one(self):
+        """
+        Pop a single element of the sequence
+        :return: True when the pop was successful, False when the end of the sequence has been reached
+        """
+        self.current_action_index += 1
```

Comment:
Because this  statement is not encapsulated by your `if` block, consecutive calls to `pop_one()` would result in a weird `current_action_index` value.
Although it does not make sense to just call `pop_one()` a lot of times, I would still prefer the index incrementation to be protected

Reply:
Fixed by using ´>=´ in the condition below